schema/workflow.yaml

```
@@ -1292,6 +1292,12 @@ $defs:
           - title: ExpressionDataSchema
             $ref: '#/$defs/runtimeExpression'
             description: An expression based event data schema.
+      data:
+        title: EventData
+        description: The event's payload data
+        oneOf:
+          - $ref: '#/$defs/runtimeExpression'
+          - type: object 
```

Actually data can be anything according to the spec: https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#event-data

```suggestion
        oneOf:
          - $ref: '#/$defs/runtimeExpression'
          - {}
```

---

Which we should also update in the dsl-reference.md

---

hmm, {} its not working

---

It appears object is more inclusive than {}
@cdavernas once you put object, additional properties is implicitly true for that object

---

You should remove the `oneOf` and should not specify the `type` at all, like so:

```
data:
  title: EventData
  description: The event's payload data
```

---

@cdavernas you are right. See my latest proposal

---

> Which we should also update in the dsl-reference.md

@fjtirado Can you also update the  `dsl-reference.md` accordingly, to specify that data can be `any`?

---

> @cdavernas you are right. See my latest proposal. Its equivalent to admit anything, but we stress the possibility of using a runtime expression in the schema itself.

Great! And it's probably more explicit! Well done!